File: Booky.PL/wwwroot/js/product.js
function loadDataTable() {
    dataTable = $('#tblData').DataTable({
        "ajax": {
            url: 'product/getAll',
            type: 'GET',
            dataType: 'Json',
            contentType: 'application/json'
        },
        "columns":
            [
                { data: 'isbn'},
                { data: 'title'},
                { data: 'description'},
                { data: "author"},
                { data: 'category.name'},
                { data: 'listPrice'},
                { data: 'price'},
                { data: 'price50'},
                { data: 'price100' },
                
                //imageURL

                {
                    data: 'imageURL',

                    render: function (data, type, row) {
                        return '<img src="/files/images/'+data+'" alt="Product Image" class="img-thumbnail" />';
                    }
                },

                {
                    data: null,

                    render: function (row) {
                        return '<div class="w-75 btn-group" role="group"> ' +
                            '<a href="/Admin/Product/Edit/' + row.id + '" class="btn btn-primary mx-2"><i class="bi bi-pencil-square"></i>Edit</a> ' +
                            '<a href="/Admin/Product/Delete/' + row.id + '" class="btn btn-danger mx-2"><i class="bi bi-trash3"></i>Delete</a> ' +
                            '</div>';
                    }
                }

            ]
    });
}

File: Booky.DAL/Models/Product.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Booky.DAL.Models
{
    public class Product
    {
        [Key]
        public int Id { get; set; }

        [Required(ErrorMessage = "Title is required")]
        public string Title { get; set; }

        public string Description { get; set; }

        [Required(ErrorMessage = "ISBN is required")]
        public string ISBN { get; set; } 

        [Required(ErrorMessage = "Author is required")]
        public string Author { get; set; }

        [Required]
       
        public double ListPrice { get; set; }

        [Required]
        
        public double Price { get; set; }

        [Required]
        
        public double Price50 { get; set; }

        [Required]
        
        public double Price100 { get; set; }

        [ForeignKey("Category")]
        public int CategoryId { get; set; }

        public Category? Category { get; set; }

        public string? ImageURL { get; set; }
    }
}
